community/consistency-check/src/main/java/org/neo4j/consistency/checking/full/MandatoryProperties.java

```diff
@@ -131,19 +121,43 @@ public MandatoryProperties( StoreAccess storeAccess )
         };
     }
 
-    private boolean safeHasNext( Iterator<?> iterator )
+    /**
+     * The iterators returned by the SchemaStorage will throw a `MalformedSchemaRuleException` wrapped in a
+     * `RuntimeException` if it encounters malformed schema rules. As the iterator implementation is prefetching this
+     * will manifest in the hasNext() call. Attempting again will actually attempt to read the schema rule after the
+     * malformed one. This method will therefore execute `hasNext()` until 1) it reads a complete schema rule, 2) all
+     * schema rules are read, or 3) it get's an exception that is not wrapping a `MalformedSchemaRuleException`.
+     */
+    private Iterable<PropertyConstraintRule> constraintsIgnoringMalformed( SchemaStorage schemaStorage )
     {
-        for (; ; )
-        {
-            try
-            {
-                return iterator.hasNext();
-            }
-            catch ( Exception e )
+        final Iterator<PropertyConstraintRule> ruleIterator = schemaStorage.schemaRules( BASE_RULE );
+        return () -> new Iterator<PropertyConstraintRule>()
             {
-                // ignore
-            }
-        }
+                @Override
+                public boolean hasNext()
+                {
+                    while ( true )
+                    {
+                        try
+                        {
+                            return ruleIterator.hasNext();
+                        }
+                        catch ( Exception e )
+                        {
+                            if ( !MalformedSchemaRuleException.class.isInstance( e.getCause() ) )
```

Comment:
No particular reason... I just took whatever came first. I had some a though telling me that getCause() gives `null` every now and then which would be a reason not to call method on it, but `instanceof` is null-safe so that's not really a valid argument. Do you see any reason to change it now apart from code consistency?